src/couch/src/couch_doc.erl

```diff
@@ -129,11 +129,24 @@ from_json_obj_validate(EJson) ->
     Doc = from_json_obj(EJson),
     case erlang:external_size(Doc#doc.body) =< MaxSize of
         true ->
+             validate_attachment_sizes(Doc#doc.atts),
              Doc;
         false ->
             throw({request_entity_too_large, Doc#doc.id})
     end.
 
+
+validate_attachment_sizes([]) ->
+    ok;
+validate_attachment_sizes(Atts) ->
+    MaxAttSize = couch_att:max_attachment_size(),
+    [begin
```

Comment:
why not `foreach`?

Reply:
Ah no good reason. Just defaulted to LC for iteration. I'll try foreach, it might look better as it avoids the begin / end bits.

Reply:
@eiri updated with foreach, looks better. Thanks!